examples/riscv/family.py

```diff
@@ -0,0 +1,81 @@
+
+from peak import family
+
+from ast_tools.passes import remove_asserts
+
+
+# A bit of hack putting the def of word and idx here
+# and not isa but it makes life easier
+class _RiscFamily_mixin:
+    @property
+    def Word(self):
+        return self.BitVector[32]
+
+    @property
+    def Idx(self):
+        return self.BitVector[5]
+
+
+class PyFamily(_RiscFamily_mixin, family.PyFamily):
+    def get_register_file(fam_self):
+        class RegisterFile:
+            def __init__(self):
+                self.rf = {fam_self.Idx(0): fam_self.Word(0)}
+
+            def _load(self, idx):
+                if not isinstance(idx, fam_self.Idx):
+                    raise TypeError(idx)
+                return self.rf[idx]
+
+            load1 = _load
+            load2 = _load
+
+            def store(self, idx, value):
+                if not isinstance(idx, fam_self.Idx):
+                    raise TypeError(idx)
+                elif not isinstance(value, fam_self.Word):
+                    raise TypeError(value)
+                elif idx != 0:
+                    self.rf[idx] = value
+        return RegisterFile
+
+
+class SMTFamily(_RiscFamily_mixin, family.SMTFamily):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._passes = remove_asserts(), *self._passes
```

Comment:
We could add this to default `SMTFamily` but we would also need to do the same for magma.  In any event it allows asserts to be in the code.

Reply:
Seems reasonable to add that by default to SMT and Magma

Reply:
I will do this after the paper push.  There is not a great way to do this right now but if we move to sequential2 something I plan to do in the feature this would not be hard.